PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public enum ObjectType
+{
+    Player,
+    Castle,
+    Enemy
+}
+public class UniversalHealth : MonoBehaviour
+{
+    [SerializeField] private int _maxHealth;
+    [SerializeField] private int _currentHealth;
+    private bool _isRegenerating = false;
+    private bool _didDamage = false;
+    private Transform _currentTransform;
+    public ObjectType _objectType;
+    private int _onePercentOfCastleHPInt;
+    
+    private void Awake()
+    {
+        _currentHealth = _maxHealth;
+        float onePercentOfCastleHP = _maxHealth * 0.005f;
+        _onePercentOfCastleHPInt = Mathf.RoundToInt(onePercentOfCastleHP);
+    }
+
+    public void TakeDamage(int amount)
+    {
+        _currentHealth -= amount;
+        if (_currentHealth <= 0)
+            Die();
+    }
+
+    public void HealHealth(int amount)
+    {
+        _currentHealth += amount;
+        if (_currentHealth >= _maxHealth)
+            _currentHealth = _maxHealth;
+    }
+
+    public void PassiveRegenHealth(int amount)
```

Comment:
Hoewel ik passive health regen een goed idee vindt zou ik het niet in het health script zelf doen, aangezien we dit waarschijnlijk liever alleen op de speler hebben.